src/cpp/session/modules/build/SessionInstallRtools.cpp

```diff
@@ -119,20 +97,40 @@ Error installRtools()
       return error;
 
    if (version == "4.2")
-      url = resolveRtools42InstallerUrl(url);
+   {
+     Error error = r::exec::RFunction(".rs.findRtools42Installer")
+            .addParam("url", url)
+            .addParam("fallbackUrl", kFallbackUrl)
+            .call(&url);
+     if (error)
+     {
+        LOG_ERROR(error);
+        url = kFallbackUrl;
+     }
+   }
 
    // form path to destination file
    std::string rtoolsBinary = url.substr(url.find_last_of('/') + 1);
    FilePath installerPath = tempPath.completeChildPath(rtoolsBinary);
    std::string destfile = string_utils::utf8ToSystem(installerPath.getAbsolutePath());
 
+   // The Rtools installer can be a large file, so we want to increase the timeout option if less than 5 min
+   // and respect the user's original configured timeout settings
+   int originalTimeoutOption = r::options::getOption<int>("timeout");
+   error = r::options::setOption<int>("timeout", std::max(300, originalTimeoutOption));
```

Comment:
I think we should just use a very large timeout here (presumedly the user would cancel the download manually if no progress were being made)

```suggestion
   error = r::options::setOption<int>("timeout", std::max(3600, originalTimeoutOption));
```

Reply:
Agreed, seems reasonable